### ISSUE TITLE
How to setup image preproccessing configs for accuracy-check 

### ISSUE THREAD

**Issue by HeywardLiu** [openvinotoolkit/openvino#14411]
##### System information (version)
- OpenVINO=> 2022.03 installed from PyPI
- Operating System / Platform => ubuntu server 20.04.01 (5.4.0-126-generic)
- Compiler => gcc 9.4.0
- Problem classification => Model has been converted from onnx successfully, but inference results seems to be wrong.
 
### UPDATE
I followed this [this page](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/swin-tiny-patch4-window7-224/model.yml#L42) which show how `swin-tiny` were converted from `onnx` to `IR` via built-in `mo` srcipt. 

I figured out that I should embed input normalziation to my `IR` model.

```
# Embed normalization to model
# Nomalize input tensor with Imagenet-1k-mean &  Imagenet-1k-std
$ mo --input_model deit_tiny_distilled_patch16_224.onnx \
     --mean_values [123.675,116.28,103.53] --scale_values [58.395,57.12,57.375] \
     --reverse_input_channels
```
I also refered to [annotation file](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/swin-tiny-patch4-window7-224/accuracy-check.yml).

With the following configs for `accuracry-check`, the accuracy grew to **18%**, however it were still much lower than my expectation.

I'm wondering how to set preprocessing configs appropriately to achieve model's expected accuracy.
```
models:
  - name: deit_tiny_distilled_patch16_224.
 
    launchers:
      - framework: dlsdk
        device: CPU
        adapter: classification
 
    datasets:
      - name: ImageNet-1k-classification
        # data_source: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet
        data_source: /home/heyward-lab/.imagenet-val
        annotation_conversion:
          converter: imagenet
          # annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet/val.txt
          annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/imagenet-val.txt
        reader: pillow_imread
 
        preprocessing:
          - type: resize
            size: 256
            aspect_ratio_scale: greater
            use_pillow: True
          - type: crop
            size: 224
            use_pillow: True
          - type: bgr_to_rgb
 
        metrics:
          - name: accuracy@top1
            type: accuracy
            top_k: 1
 
          - name: accuracy@top5
            type: accuracy
            top_k: 5
```
---
---
Hi community,

I conveted a image classification model [`deit-tiny-distilled-patch16-224`](https://github.com/facebookresearch/deit/blob/main/README_deit.md) through the following steps
 - load it from torch-hub (`torch`)
 - export it from torch to onnx through the following script (`torch` -> `onnx`)
```
import torch
import timm

# now load it with torchhub
pt_model = torch.hub.load('facebookresearch/deit:main', 'deit_tiny_distilled_patch16_224', pretrained=True)

# Convert to onnx 
dummy_input = torch.randn((1, 3, 224, 224))
pt_model.eval()
torch.onnx.export(
    model = pt_model,
    args = dummy_input,                           
    f = "deit_tiny_patch16_224.onnx",
    verbose = True,             
    opset_version=12
)  
```
 - finally converted onnx to IR via openvino built-in `model optimizer`  (`onnx` -> `openvino IR (.xml)`)

But when I tired to run `hello_classification` and `accuracy-check` sample scripts, the accuracy of models seems to be corrupted. The accuracy measured by `accuracy-chek` dropped to 0.

I've tried the following things to figure out where's going wroing
1. `onnx` models was fine. I've converted my onnx model to tensorflow and it worked fine. It's accruracy was same as pytorch model.   

4. Other models converted via `torch` -> `onnx` -> `IR` path had no issuues. I've also tried `swin-tiny-patch4-window7` which was downloaded from openvino `omz_downloader` and it worked perfectly. The accurarcy measured by `accuracy-check` tool was almost same as pytorch model.

Every srcipts mentioned above run correctly without warnings or error messages. How can resolve this problem ? 

---

##### Steps to reproduce
- Convert onnx model via `mo` 
```
mo --input_model ~/deit_proj/model_converter/models/onnx_models/onnxsim_deit_tiny_distilled_patch16_224.onnx 
/home/heyward-lab/vino_to_tf/vino/lib/python3.9/site-packages/openvino/offline_transformations/__init__.py:10: FutureWarning: The module is private and following namespace `offline_transformations` will be removed in the future, use `openvino.runtime.passes` instead!
  warnings.warn(message="The module is private and following namespace "
[ INFO ] The model was converted to IR v11, the latest model format that corresponds to the source DL framework input/output format. While IR v11 is backwards compatible with OpenVINO Inference Engine API v1.0, please use API v2.0 (as of 2022.1) to take advantage of the latest improvements in IR v11.
Find more information about API v2.0 and IR v11 at https://docs.openvino.ai
[ SUCCESS ] Generated IR version 11 model.
[ SUCCESS ] XML file: /home/heyward-lab/vino_to_tf/accuracy_check_files/onnxsim_deit_tiny_distilled_patch16_224.xml
[ SUCCESS ] BIN file: /home/heyward-lab/vino_to_tf/accuracy_check_files/onnxsim_deit_tiny_distilled_patch16_224.bin
```

---

##### Run inference on ILSVRC-2012-val samples on `hello_classification.py`

- ILSVRC_2012_val_0000001, ground-truth label = 65
![ILSVRC2012_val_00000001](https://user-images.githubusercontent.com/54396044/205678389-d693d534-c041-4a65-8fb2-13e37951c592.JPEG)

- swin-tiny-patch4-window7 
```
 python3 hello_classification.py ir_models/swin-tiny-patch4-window7-224/FP32/swin-tiny-patch4-window7-224.xml ~/imagenet-val/ILSVRC2012_val_00000001.JPEG CPU
[ INFO ] Creating OpenVINO Runtime Core
[ INFO ] Reading the model: ir_models/swin-tiny-patch4-window7-224/FP32/swin-tiny-patch4-window7-224.xml
[ INFO ] Loading the model to the plugin
[ INFO ] Starting inference in synchronous mode
[ INFO ] Image path: /home/heyward-lab/imagenet-val/ILSVRC2012_val_00000001.JPEG
[ INFO ] Top 10 results: 
[ INFO ] class_id probability
[ INFO ] --------------------
[ INFO ] 65       10.6597443
[ INFO ] 54       5.9004016
[ INFO ] 58       5.4193702
[ INFO ] 67       4.5983825
[ INFO ] 68       4.4895983
[ INFO ] 62       4.1857924
[ INFO ] 60       3.7454436
[ INFO ] 63       3.6236312
[ INFO ] 390      3.6202765
[ INFO ] 66       3.5734475
[ INFO ] 
[ INFO ] This sample is an API example, for any performance measurements please use the dedicated benchmark_app tool
```
- deit-tiny-distilled-patch16-224
```
python3 hello_classification.py ir_models/deit_tiny_disilled_patch16_224/FP16/deit_tiny_distilled_patch16_224.xml ~/imagenet-val/ILSVRC2012_val_00000001.JPEG CPU
[ INFO ] Creating OpenVINO Runtime Core
[ INFO ] Reading the model: ir_models/deit_tiny_disilled_patch16_224/FP16/deit_tiny_distilled_patch16_224.xml
[ INFO ] Loading the model to the plugin
[ INFO ] Starting inference in synchronous mode
[ INFO ] Image path: /home/heyward-lab/imagenet-val/ILSVRC2012_val_00000001.JPEG
[ INFO ] Top 10 results: 
[ INFO ] class_id probability
[ INFO ] --------------------
[ INFO ] 324      4.7675791
[ INFO ] 470      4.6060848
[ INFO ] 862      4.5437469
[ INFO ] 818      4.3673725
[ INFO ] 969      4.2719889
[ INFO ] 793      4.0580430
[ INFO ] 606      4.0274611
[ INFO ] 922      4.0164814
[ INFO ] 529      3.9683518
[ INFO ] 181      3.9012418
[ INFO ] 
[ INFO ] This sample is an API example, for any performance measurements please use the dedicated benchmark_app tool
```

- ILSVRC_2012_val_0000002.JPEG , ground truth label = 970
![ILSVRC2012_val_00000002](https://user-images.githubusercontent.com/54396044/205679747-c12ae9dc-c6f2-4885-b837-0211b87ddd21.JPEG)
- swin
```
python3 hello_classification.py ir_models/swin-tiny-patch4-window7-224/FP32/swin-tiny-patch4-window7-224.xml ~/imagenet-val/ILSVRC2012_val_00000002.JPEG CPU
[ INFO ] Creating OpenVINO Runtime Core
[ INFO ] Reading the model: ir_models/swin-tiny-patch4-window7-224/FP32/swin-tiny-patch4-window7-224.xml
[ INFO ] Loading the model to the plugin
[ INFO ] Starting inference in synchronous mode
[ INFO ] Image path: /home/heyward-lab/imagenet-val/ILSVRC2012_val_00000002.JPEG
[ INFO ] Top 10 results: 
[ INFO ] class_id probability
[ INFO ] --------------------
[ INFO ] 795      9.9081135
[ INFO ] 970      7.1770492
[ INFO ] 796      3.3898046
[ INFO ] 672      3.2183545
[ INFO ] 872      2.6605642
[ INFO ] 979      2.6305120
[ INFO ] 521      2.3828297
[ INFO ] 537      2.2860665
[ INFO ] 647      2.2370093
[ INFO ] 915      2.1211185
[ INFO ] 
[ INFO ] This sample is an API example, for any performance measurements please use the dedicated benchmark_app tool
```

- deit-tiny
```
python3 hello_classification.py ir_models/deit_tiny_disilled_patch16_224/FP16/deit_tiny_distilled_patch16_224.xml ~/imagenet-val/ILSVRC2012_val_00000002.JPEG CPU
[ INFO ] Creating OpenVINO Runtime Core
[ INFO ] Reading the model: ir_models/deit_tiny_disilled_patch16_224/FP16/deit_tiny_distilled_patch16_224.xml
[ INFO ] Loading the model to the plugin
[ INFO ] Starting inference in synchronous mode
[ INFO ] Image path: /home/heyward-lab/imagenet-val/ILSVRC2012_val_00000002.JPEG
[ INFO ] Top 10 results: 
[ INFO ] class_id probability
[ INFO ] --------------------
[ INFO ] 862      4.8931828
[ INFO ] 470      4.5322757
[ INFO ] 324      4.4012027
[ INFO ] 818      4.3124208
[ INFO ] 969      4.2759705
[ INFO ] 173      4.1416321
[ INFO ] 452      4.0997410
[ INFO ] 181      3.9916825
[ INFO ] 870      3.8489528
[ INFO ] 793      3.7448819
[ INFO ] 
[ INFO ] This sample is an API example, for any performance measurements please use the dedicated benchmark_app tool
```
---
##### Run `accuracy-check` on ImageNet-val (50k images)
- swin-tiny
```
accuracy_check -c swin-tiny-acc-check.yml -m ir_models/swin-tiny-patch4-window7-224/FP32/

Processing info:
model: swin-tiny-patch4-window7-224
launcher: openvino
device: CPU
dataset: classification_dataset
OpenCV version: 4.6.0
Annotation conversion for classification_dataset dataset has been started
Parameters to be used for conversion:
converter: imagenet
annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/imagenet-val.txt
Annotation conversion for classification_dataset dataset has been finished
2022-12-05 08:44:14.396499: I tensorflow/core/util/util.cc:169] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
IE version: 2022.3.0-8507-a5a969886be
Loaded CPU plugin version:
    CPU - openvino_intel_cpu_plugin: 2022.3.2022.3.0-8507-a5a969886be
Found model ir_models/swin-tiny-patch4-window7-224/FP32/swin-tiny-patch4-window7-224.xml
Found weights ir_models/swin-tiny-patch4-window7-224/FP32/swin-tiny-patch4-window7-224.bin
Input info:
        Node name: input
        Tensor names: input
        precision: f32
        shape: (1, 3, 224, 224)

Output info
        Node name: probs/sink_port_0
        Tensor names: probs
        precision: f32
        shape: (1, 1000)
        
50000 objects processed in 1466.410 seconds
accuracy@top1: 80.92%
accuracy@top5: 95.42%
```
- deit-tiny
  - It's simply acc@1=0%, acc@5 = 0%
  - it seems to be corrupted

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by HeywardLiu** [comment]
### Update
Hi, 

I've further converted [`TinyViT`](https://github.com/microsoft/Cream/tree/main/TinyViT) pytorch models to openvino IR. It had same issue as `deit`. 

I found that converted a pytorch model only infer as expected when downloaded from `omz_downloader`.

So I decided to take a look on `swin-tiny-patch4-window7` pytorch models in [open_model_zoo](https://github.com/openvinotoolkit/openvino/issues/14411). 

I noticed that the order of color channels was `BGR` instead of `RGB`. 

![image](https://user-images.githubusercontent.com/54396044/205733338-2273c057-e4f9-439b-9aef-a61cb0a2c884.png)

---
But I've used `bgr_to_rgb` for data-preprocessing in my `.yaml` file (shown below) for accuracy-check

```
models:
  - name: tiny_vit_5m_224
 
    launchers:
      - framework: dlsdk
        device: CPU
        adapter: classification
 
    datasets:
      - name: classification_dataset
        # data_source: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet
        data_source: /home/heyward-lab/imagenet-val
        annotation_conversion:
          converter: imagenet
          # annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet/val.txt
          annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/imagenet-val.txt
        reader: pillow_imread
 
        preprocessing:
          - type: resize
            size: 256
            aspect_ratio_scale: greater
            use_pillow: True
          - type: crop
            size: 224
            use_pillow: True
          - type: bgr_to_rgb
 
        metrics:
          - name: accuracy@top1
            type: accuracy
            top_k: 1
 
          - name: accuracy@top5
            type: accuracy
            top_k: 5
```
---
In addition, I used data-preprocessing config for `swin-tiny` and its worked fine. 

So I'm confused now. Didn't `swin` expect `BGR` ordered input? Why would `RGB` ordered also infered perfectly? 

- swin-tiny / type: bgr_to_rgb / acc@1 = 81%
```
models:
  - name: swin-tiny-patch4-window7-224
 
    launchers:
      - framework: dlsdk
        device: CPU
        adapter: classification
 
    datasets:
      - name: classification_dataset
        data_source: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet
        # data_source: /home/heyward-lab/imagenet-val
        annotation_conversion:
          converter: imagenet
          annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet/val.txt
          # annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/imagenet-val.txt
        reader: pillow_imread
 
        preprocessing:
          - type: resize
            size: 256
            aspect_ratio_scale: greater
            use_pillow: True
          - type: crop
            size: 224
            use_pillow: True
          - type: bgr_to_rgb
 
        metrics:
          - name: accuracy@top1
            type: accuracy
            top_k: 1
 
          - name: accuracy@top5
            type: accuracy
            top_k: 5
```

**Comment by HeywardLiu** [comment]
### UPDATE
I followed this [this page](https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/swin-tiny-patch4-window7-224/model.yml#L42) which show how `swin-tiny` were converted from `onnx` to `IR` via built-in `mo` srcipt. 

I figured out that I should embed input normalziation to my `IR` model.

```
# Embed normalization to model
# Nomalize input tensor with Imagenet-1k-mean &  Imagenet-1k-std
$ mo --input_model deit_tiny_distilled_patch16_224.onnx \
     --mean_values [123.675,116.28,103.53] --scale_values [58.395,57.12,57.375] \
     --reverse_input_channels
```

With the following annotations file for `accuracry-check`, the accuracy were still much lower than my expectation.

```
models:
  - name: deit_tiny_distilled_patch16_224.
 
    launchers:
      - framework: dlsdk
        device: CPU
        adapter: classification
 
    datasets:
      - name: ImageNet-1k-classification
        # data_source: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet
        data_source: /home/heyward-lab/.imagenet-val
        annotation_conversion:
          converter: imagenet
          # annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/MiniImageNet/val.txt
          annotation_file: /home/heyward-lab/vino_to_tf/accuracy_check_files/imagenet-val.txt
        reader: pillow_imread
 
        preprocessing:
          - type: resize
            size: 256
            aspect_ratio_scale: greater
            use_pillow: True
          - type: crop
            size: 224
            use_pillow: True
          - type: bgr_to_rgb
 
        metrics:
          - name: accuracy@top1
            type: accuracy
            top_k: 1
 
          - name: accuracy@top5
            type: accuracy
            top_k: 5
```

**Comment by mbencer** [comment]
I am confirming that ONNX->IR conversion (and IR inference) looks correctly. @eaidova Could you take a look and help with a proper accuracy checker configuration?

**Comment by eaidova** [comment]
just for my clarifiaction, @mbencer about which model do you speak swin-tiny or deit-tiny?
@HeywardLiu, yes, open model zoo config examples and openvino samples expects that normalization embedded to the model. Could you please check that you have correct results using pytorch? Such small results if model valid can speak that model weights are not loaded

**Comment by eaidova** [comment]
@HeywardLiu  which openvino version do you use?
I tried single image inference for both pytorch and openvino 2022.3 according to notebook from README.md
here my results: https://gist.github.com/eaidova/98a5312c163458a7bdb1998033049253
Looks like in 2022.3 model works correctly, could you please check with this version?

Regarding preprocessing configuration, according to this sample and taking to account that you embed preprocessing, it looks correct and suitable for deit model

**Comment by mbencer** [comment]
> just for my clarifiaction, @mbencer about which model do you speak swin-tiny or deit-tiny? @HeywardLiu, yes, open model zoo config examples and openvino samples expects that normalization embedded to the model. Could you please check that you have correct results using pytorch? Such small results if model valid can speak that model weights are not loaded

I mean `deit_tiny_patch16_224.onnx`

**Comment by andrei-kochin** [comment]
@HeywardLiu do you have any further questions or clarifications?

**Comment by andrei-kochin** [comment]
Closing. Feel free to-reopen if there any further queries.
